cfme/tests/automate/test_method.py

```diff
@@ -348,3 +350,99 @@ def test_send_email_method(smtp_test, klass):
+        if definition.exists:
+            definition.delete()
+
+
+@pytest.fixture
+def service_req(request, generic_catalog_item):
+    # Creating generic service
+    service_request = generic_catalog_item.appliance.rest_api.collections.service_templates.get(
+        name=generic_catalog_item.name
+    ).action.order()
+    request.addfinalizer(service_request.action.delete)
+
+
+@pytest.mark.tier(1)
+def test_automate_generic_object_service_associations(appliance, klass, service_req,
+                                                      generic_definition):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/10h
+        caseimportance: medium
+        startsin: 5.7
+        casecomponent: Automate
+
+    Bugzilla:
+        1410920
+    """
+    # Ruby code
+    script = 'go_class = $evm.vmdb(:generic_object_definition).find_by(:name => "{name}")\n'.format(
+        name=generic_definition.name
+    )
+    script = script + (
+        'load_balancer = go_class.create_object(:name => "Test Load Balancer", :location => '
+        '"Mahwah")\n'
+        '$evm.log("info", "XYZ go object: #{load_balancer.inspect}")\n'
+        'service = $evm.vmdb(:service).first\n'
+        'load_balancer.services = [service]\n'
+        'content_type = "message"\n'
+        'load_balancer.save!\n'
+        '$evm.log("info", "XYZ load balancer got service: #{load_balancer.services.first.inspect}")'
+        '\nexit MIQ_OK'
+    )
+    with appliance.context.use(ViaUI):
+        # Adding schema for executing method
+        klass.schema.add_fields({'name': 'execute', 'type': 'Method', 'data_type': 'String'})
+
+        # Adding method
+        method = klass.methods.create(
+            name=fauxfactory.gen_alphanumeric(),
+            display_name=fauxfactory.gen_alphanumeric(),
+            location='inline',
+            script=script)
+
+        # Adding instance to call automate method
+        instance = klass.instances.create(
+            name=fauxfactory.gen_alphanumeric(),
+            display_name=fauxfactory.gen_alphanumeric(),
+            description=fauxfactory.gen_alphanumeric(),
+            fields={'execute': {'value': method.name}}
+        )
+
+        result = LogValidator(
+            "/var/www/miq/vmdb/log/automation.log",
+            matched_patterns=[
+                r'.*XYZ go object: #<MiqAeServiceGenericObject.*',
+                r'.*XYZ load balancer got service: #<MiqAeServiceService.*'
+            ],
+        )
+        result.fix_before_start()
+
+        # Executing automate method using simulation
+        simulate(
+            appliance=klass.appliance,
+            attributes_values={
+                "namespace": klass.namespace.name,
+                "class": klass.name,
+                "instance": instance.name,
+            },
+            message="create",
+            request="Call_Instance",
+            execute_methods=True,
+        )
+        result.validate_logs()
```

Comment:
Can you tell how are we asserting on go association with service . Can we check the service page and see if GO is displayed there ?

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1410920#c3
```
In the tail'ed log:
There should be no ERROR lines related to the execution.
There should be these two lines:
<AEMethod gotest> XYZ go object: #<MiqAeServiceGenericObject ....something...>
<AEMethod gotest> XYZ load balancer got service: #<MiqAeServiceService:....something....>
If there is "XYZ load balancer got service: nil", then this bug was reproduced.
```
On the basis of comment, I am verifying logs.

I am not much familiar with generic objects. So please let me know other ways to improve implementation of this test case.